legend-engine-authentication-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/authentication/ConnectionProvider.java

```diff
@@ -0,0 +1,23 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package org.finos.legend.engine.plan.execution.authentication;
+
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.runtime.connection.authentication.AuthenticationSpec;
+import org.finos.legend.engine.shared.core.identity.Identity;
+
+public abstract class ConnectionProvider<T> {
+
+    public abstract T makeConnection(ConnectionSpec connectionSpec, AuthenticationSpec authenticationSpec, Identity identity) throws Exception;
```

Comment:
possibly we should have  authSpec -> cred and conn -> physical conn slow mention here itself at generic level